src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,73 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
 public class WorkWithFile {
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeDataToFile(calculateStatistics(fromFileName), toFileName);
+    }
+
+    private String calculateStatistics(String fromFileName) {
+        final int AcT = 0;
+        final int DigiT = 1;
+        int supply = 0;
+        int buy = 0;
+        int result = 0;
+        String [] temp = readFromFile(fromFileName);
+
+        for (int i = 0; i < temp.length; i++) {
+            String [] splitLine = temp[i].split(",");
+            switch (splitLine[AcT]) {
+                case "supply":
+                    supply += Integer.parseInt(splitLine[DigiT]);
+                    break;
+                case "buy":
+                    buy += Integer.parseInt(splitLine[DigiT]);
+                    break;
+                default:
+                    break;
+            }
+        }
+        result = supply - buy;
+        return supply + " " + buy + " " + result;
+    }
```

Comment:
The method `calculateStatistics` seems to be doing multiple things: reading from a file, processing the data, and creating a result string. Consider refactoring this method to adhere to the Single Responsibility Principle. You could, for example, have one method to read and process the data, and another to format the result.